tools/ReadConsoleInputStream/NativeMethods.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Nivot.Terminal
+{
+    internal static class NativeMethods
+    {
+        private static int MakeHRFromErrorCode(int errorCode)
+        {
+            // Don't convert it if it is already an HRESULT
+            if ((0xFFFF0000 & errorCode) != 0)
+                return errorCode;
```

Comment:
For 1, I'm surprised that such a method doesn't already exist in C# somewhere. But I couldn't find one!
For 2, I'd be inclined to take the opportunity to make this throw/fail if it is used to try to convert something with the high order bits set as it would indicate a programmer error to HRESULT-ify something that's already HRESULT/NTSTATUS. I know the original HRESULT_FROM_WIN32 C macro wasn't intelligent, but we might want to set a better example.

Reply:
I could do this, or I could inline the GetLastWin32Error method into the calling method and then we don't have to worry about this.

Reply:
I'll put in a debug assert also.

Reply:
works for me